src/components/UploadFile.tsx

```diff
@@ -0,0 +1,162 @@
+
+    const handleFileUpload = (files: FileObject[]) => {
+        const totalSize = files.reduce((sum, file) => sum + (file.size ?? 0), 0);
+
+        if (totalFilesSizeLimit) {
+            if (totalSize > totalFilesSizeLimit) {
+                setError(translate('attachmentPicker.sizeExceededWithValue', {maxUploadSizeInMB: totalFilesSizeLimit / (1024 * 1024)}));
+                return;
+            }
+        }
+
+        if (fileLimit) {
+            if (files.length > 0) {
+                if (files.length > fileLimit) {
+                    setError(translate('attachmentPicker.tooManyFiles', {fileLimit}));
+                    return;
+                }
+            }
+        }
+
+        if (acceptedFileTypes.length > 0) {
+            const filesExtensions = files.map((file) => FileUtils.splitExtensionFromFileName(file?.name ?? '').fileExtension.toLowerCase());
+
+            if (acceptedFileTypes.every((element) => !filesExtensions.includes(element as string))) {
+                setError(translate('attachmentPicker.notAllowedExtension'));
+                return;
+            }
+        }
+
+        onInputChange(files);
+        onUpload(files);
+    };
+
+    return (
+        <View style={[styles.alignItemsStart, style]}>
+            <AttachmentPicker
+                acceptedFileTypes={acceptedFileTypes}
+                fileLimit={fileLimit}
+                allowMultiple={!!fileLimit}
+            >
+                {({openPicker}) => (
+                    <Button
+                        medium
+                        text={buttonText}
+                        accessibilityLabel={buttonText}
+                        onPress={() => {
+                            openPicker({
+                                onPicked: handleFileUpload,
+                            });
+                        }}
+                    />
+                )}
+            </AttachmentPicker>
+            {uploadedFiles.map((file) => (
+                <View
+                    style={[styles.flexRow, styles.alignItemsCenter, styles.justifyContentCenter, styles.border, styles.p4, styles.mt3]}
+                    key={file.uri}
+                >
+                    <Icon
+                        height={variables.iconSizeNormal}
+                        width={variables.iconSizeSemiSmall}
+                        src={Expensicons.Paperclip}
+                        fill={theme.textSupporting}
+                    />
+                    <Text style={[styles.ml4, styles.mr3, styles.textBold]}>{file.name}</Text>
+                    <PressableWithFeedback
+                        onPress={() => onRemove(file?.uri ?? '')}
+                        role={CONST.ROLE.BUTTON}
+                        accessibilityLabel={translate('common.remove')}
+                    >
+                        <Icon
+                            src={Expensicons.Close}
+                            fill={theme.textSupporting}
+                            small
```

Comment:
Not sure what size this prop actually is but I think it should be 20x20

Reply:
updated